frontend/src/metabase/visualizations/components/ChartSettings/ChartSettings.tsx

```diff
@@ -39,46 +45,75 @@ import {
 } from "./ChartSettings.styled";
 import type {
   ChartSettingsProps,
-  ChartSettingsWithStateProps,
+  ChartSettingsVisualizationProps,
+  DashboardChartSettingsProps,
+  QuestionChartSettingsProps,
   Widget,
 } from "./types";
 
 // section names are localized
 const DEFAULT_TAB_PRIORITY = [t`Data`];
 
-export const ChartSettings = ({
-  initial,
-  settings: propSettings,
-  series,
-  computedSettings: propComputedSettings,
-  onChange,
-  isDashboard = false,
-  noPreview = false,
+const ChartSettingsVisualization = ({
   dashboard,
   dashcard,
+  onCancel,
   onDone,
-  onClose,
-  question,
-  className,
-  widgets: propWidgets,
-}: ChartSettingsProps) => {
-  const [currentSection, setCurrentSection] = useState<string | null>(
-    initial?.section ?? null,
-  );
-  const [currentWidget, setCurrentWidget] = useState<Widget | null>(
-    initial?.widget ?? null,
-  );
-  const [popoverRef, setPopoverRef] = useState<HTMLElement | null>();
-  const [warnings, setWarnings] = useState();
+  onReset,
+  onUpdateVisualizationSettings,
+  onUpdateWarnings,
+  rawSeries,
+  warnings,
+}: ChartSettingsVisualizationProps) => (
+  <ChartSettingsPreview>
+    <SectionWarnings warnings={warnings} size={20} />
+    <ChartSettingsVisualizationContainer>
+      <Visualization
+        className={CS.spread}
+        rawSeries={rawSeries}
+        showTitle
+        isEditing
+        isDashboard
+        dashboard={dashboard}
+        dashcard={dashcard}
+        isSettings
+        showWarnings
+        onUpdateVisualizationSettings={onUpdateVisualizationSettings}
+        onUpdateWarnings={onUpdateWarnings}
+      />
+    </ChartSettingsVisualizationContainer>
+    <ChartSettingsFooter
+      onDone={onDone}
+      onCancel={onCancel}
+      onReset={onReset}
+    />
+  </ChartSettingsPreview>
+);
 
-  const chartSettings = useMemo(
-    () => propSettings || series[0].card.visualization_settings,
-    [series, propSettings],
-  );
+type UseChartSettingsStateParams = {
+  settings?: VisualizationSettings;
+  series: any;
+  onChange: any;
```

Comment:
Do we have a better type for `onChange`? Is this because we don't have a properly-typed `series`?